tests/core/tests/test_resources.py

```diff
@@ -2570,3 +2571,143 @@ def test_has_natural_foreign_key(self):
                 resources.has_natural_foreign_key(model),
                 expected_result
             )
+
+
+class ManyToManyWidgetDiffTest(TestCase):
+    fixtures = ["category", "book"]
+
+    def setUp(self):
+        pass
+
+    def test_m2m_on_new_entity(self):
+        # issue 1558
+        class _BookM2MResource(resources.ModelResource):
```

Comment:
This can be shortened to

```
        class _BookM2MResource(resources.ModelResource):
            class Meta:
                model = Book
```

It can also be declared at test class level to avoid repeating the declaration in each test.